Mathlib/Analysis/Normed/Module/Complemented.lean

```diff
@@ -24,10 +24,36 @@ complemented subspace, normed vector space
 variable {𝕜 E F G : Type*} [NontriviallyNormedField 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E]
   [NormedAddCommGroup F] [NormedSpace 𝕜 F] [NormedAddCommGroup G] [NormedSpace 𝕜 G]
 
-noncomputable section
-
 open LinearMap (ker range)
 
+namespace IsIdempotentElem
+
+lemma ker_id_sub_eq_range {P : E →L[𝕜] E} (h : IsIdempotentElem P) : ker (1 - P) = range P := by
```

Comment:
This would also work for `P : E →ₗ[𝕜] E` but would then need some way of inferring "P idempotent in `E →L[𝕜] E` implies P idempotent in `E →ₗ[𝕜] E`".

Reply:
One more way to write this subspace: `LinearMap.eqLocus P 1`.

Reply:
> This would also work for `P : E →ₗ[𝕜] E` but would then need some way of inferring "P idempotent in `E →L[𝕜] E` implies P idempotent in `E →ₗ[𝕜] E`".

It's just `ContinuousLinearMap.coe_injective h`.